OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -113,7 +114,8 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 				return NextActivity;
 			}
 
-			var desiredFacing = (Target.CenterPosition - self.CenterPosition).Yaw.Facing;
+			var targetedPosition = Target.Positions.PositionClosestTo(pos);
```

Comment:
Turreted units work differently again, so you also need to update `Turreted.FaceTarget`.